opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -248,84 +139,164 @@ message MetricDescriptor {
-    AGGREGATION_TEMPORALITY_CUMULATIVE = 2;
-  }
+// Gauge represents the type of a scalar metric that always exports the
+// "current value" for every data point. It should be used for an "unknown"
+// aggregation.
+// 
+// A Gauge does not support different aggregation temporalities. Given the
+// aggregation is unknown, points cannot be combined using the same
+// aggregation, regardless of aggregation temporalities. Therefore,
+// AggregationTemporality is not included. Consequently, this also means
+// "StartTimeUnixNano" is ignored for all data points.
+//
+// A Metric of this Type MUST store its values as Int64DataPoint or
+// DoubleDataPoint.
+message Gauge {
+  repeated Int64DataPoint int64_data_points = 1;
+  repeated DoubleDataPoint double_data_points = 2;
+
+  // It describes the value type of the measurement used to build this
+  // aggregation.
+  //
+  // Determines if the points are Int64DataPoint or DoubleDataPoint, as well
+  // as the value type of the exemplars.
+  MeasurementValueType measurement_value_type = 3;
+}
+
+// Sum represents the type of a numeric scalar metric that is calculated as a
+// sum of all reported measurements over a time interval.
+//
+// TODO: Decide if this should support only MonotonicSum
+// https://github.com/open-telemetry/opentelemetry-specification/issues/725.
+//
+// A Metric of this Type MUST store its values as Int64DataPoint or
+// DoubleDataPoint.
+message Sum {
+  repeated Int64DataPoint int64_data_points = 1;
+  repeated DoubleDataPoint double_data_points = 2;
+
+  // It describes the value type of the measurement used to build this
+  // aggregation.
+  //
+  // Determines if the points are Int64DataPoint or DoubleDataPoint, as well
+  // as the value type of the exemplars.
+  MeasurementValueType measurement_value_type = 3;
+  
+  // aggregation_temporality describes if the aggregator reports delta changes
+  // since last report time, or cumulative changes since a fixed start time.
+  AggregationTemporality aggregation_temporality = 4;
+
+  // If "true" means that the sum is monotonic.
+  bool is_monotonic = 5;
+}
+
+// Represents the type of a metric that is calculated by aggregating as a
+// Histogram of all reported measurements over a time interval.
+//
+// A Metric of this Type MUST store its values as HistogramDataPoint.
+message Histogram {
+  repeated HistogramDataPoint histogram_data_points = 1;
+
+  // It describes the value type of the measurement used to build this
+  // aggregation.
+  //
+  // Determines the value type of the exemplars.
+  MeasurementValueType measurement_value_type = 2;
+
+  // aggregation_temporality describes if the aggregator reports delta changes
+  // since last report time, or cumulative changes since a fixed start time.
+  AggregationTemporality aggregation_temporality = 3;
+}
+
+// MeasurementValueType determines the value type for a measurement.
+// TODO: There is an open question about whether this should control int64 vs
```

Comment:
To be clear this refers to the Sum fields. If inputs are doubles, the sum is probably a double. If inputs are integers and the numbers are large, they'll run into a loss of precision.

There's a separate question about whether Count fields can be doubles, which they would be if generating a histogram from sampled metrics (as in statsd's `@sample-rate` parameter, or if a sampled trace was converted into a metric event). 

I wish to not hold up the specification over either of these questions, so perhaps we could go with a single Double-value Histogram for now, and if ever we decide to add an Integer histogram add it and rename the existing one DoubleHistogram.

If ever we decide to embrace sampled metric events (which statsd will push us to do, I think), then having an integer count field is still OK, we can extend the struct with a double to indicate the degree of sampling that took place. Discussed here:
https://github.com/open-telemetry/opentelemetry-proto/issues/73

Reply:
#205 created to remove this TODO.